### ISSUE TITLE
realworld submission 

### ISSUE THREAD

**Issue by purpledawn777** [adamhaile/surplus#59]
Referring to issue #57 (which I am not allowed to comment on) it would be great to see a realworld submission for S/Surpplus (https://github.com/gothinkster/realworld ) : This is useful to compare frameworks and
it would more clearly communicate the strengths & caveats of the S/Surplus paradigm ~ quantitatively for example in lines of code, etc, but also so one can make a real qualitative comparison, and would make a great tutorial/starter kit on potential architectures: 
namely, how to handle nested routing, dynamically loaded modules and tree shaking module bundles ; server side render and client hydrating; fractal state, etc. 
ie, a prescriptive starter kit showing how to make real use of S/surplus that one can clone with degit and get to work with!

**Comment by adamhaile** [comment]
That's not a bad idea.  I just finished a small app, one that's about the size of the realworld demo.  I may just port over the patterns I'm using there.  I'll take a look at what other's have done.

**Comment by adamhaile** [comment]
Also, dunno why you couldn't comment on #57?  No special rights on it that I'm aware of.  If that's still the case let me know.

**Comment by purpledawn777** [comment]
Great, Looking forward to it! 
re 57, This was my first github post ever (just for this request!), so maybe I wasn't fully authenticated yet..

**Comment by purpledawn777** [comment]
Starting to look at this, looks great!
One bug I encountered though: starting out logged out, if I click on functionality needing authentication [eg add to favorites], I get redirected to the sign-in page (great) but then clicking on any of the links (eg register, or conduit/home) doesn't work /  keeps me on the sign-in page. 
On the other hand, if I explicitly go to the sign-in page directly from the home page, the links (register, etc) do work correctly.
It seems like once being redirected to the sign-in page via authMonitor.authenticate(), you stay in that mode, always redirecting to the login page until logged in.

**Comment by adamhaile** [comment]
Yeah, that's a good point.  I moved the handling of authentication into the AppContainer component, thinking that that way the individual pages wouldn't need to deal with it, but that has the result that once one page has requested login, they all get stuck in that mode.  I'll refactor that.  Thanks for taking a look!
